File: week10/week10-2.py
# week10-2.py 輾轉相除法 最大公因數
# a大的,b中的,c小的 greatest common divisor
#include <stdio.h>
def gcd(a,b):

  if a==0: return b
  if b==0: return a
  return gcd(b, a%b)

a = int(input('請輸入a:'))
b = int(input('請輸入b:'))

print('最大公因數是:',gcd(a,b))

File: week10/week10-2.cpp
/// week10-2.cpp 劣锣埃猭 程そ计
/// a,bい,c greatest common divisor
#include <stdio.h>
int gcd(int a, int b)
{
    if(a==0) return b;
    if(b==0) return a;
    return gcd( b, a%b);
}
int main()
{
    printf("叫块 a b ㄢ计:");
    int a, b;
    scanf("%d%d", &a, &b);
    printf("程そ计琌: %d\n", gcd(a, b));
}
